src/price_data.py

```diff
@@ -326,3 +328,133 @@ def get_cost(self, tr: Union[Operation, SoldCoin], reference_coin: str = config.
         if isinstance(tr, SoldCoin):
             return price * tr.sold
         raise NotImplementedError
+    
+    def get_candles(self, start: int, stop: int, symbol: str) ->list:
+        if self.exchange.has['fetchOHLCV']:
+            sleep(self.exchange.rateLimit / 1000)  # time.sleep wants seconds
+            # get 2min before and after range
+            return self.exchange.fetch_ohlcv(symbol, '1m', start-1000*60*2, max(int((stop-start)/1000/60)+5, 1))
```

Comment:
1. good point
2. Found a bug: some exchanges have a limit for how many candles you can request. Binance is at 1000 candles, i do not check in this function if the amount of candles is below 1000. The syntax for fetch ohlcv is (symbol, timeframe, start,stop) where start is the start date in ms since epoch and  stop the amount of candles to fetch) `max(int((stop-start)/1000/60)+5,1)` calculates the amount of candles needed for this timerange + an extra 5. Just noticed that in the current implementation `max` is not needed, because the minimal amount calculated is always at least 5)

3.
ccxt standartises all the apis from exchanges into one usable call. Binance only ccxt would be overkill, but if we'd support multiple exchanges ccxt would be beneficial.  `if self.exchange.has['fetchOHLCV']:` checks if the exchange supports the api call we want, haven't tested which exchanges in ccxt support this apicall but the larger supported exchanges(coinbase,binance etc) should be. fine). But we maybe should move the  `if self.exchange.has['fetchOHLCV']:` to the initialization step.

Reply:
If the maximum candle count is limited, we should adjust the timestamppairs generation. It might be beneficial if we move it to another function.

We might come up with a way to calculate the batches optimal, reducing the needed requests as much as possible.

Reply:
i couldnt get 2. fixed each exchange has a different max limit and i have jet to find a way to determine this limit(conbasepro is at 300 binance at 1000)